_episodes/08-adding-ci-to-your-code.md

```diff
@@ -83,32 +87,32 @@ git push -u origin feature/add-actions
 
 ## No root-config?
 
-Ok, so maybe we were a little naive here. GitHub runners come pre-installed with a wide variety of software that is commonly needed in CI workflows (e.g. for Ubuntu 22.04 runners the list can be found [here](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md)). ROOT is not pre-installed, so we will have to add a step to install it ourselves. After reading the ROOT documentation, we find that a convenient way to get it installed for most linux distributions is using Conda. Here's how we can do it.
+Ok, so maybe we were a little naive here. GitHub runners come pre-installed with a wide variety of software that is commonly needed in CI workflows (e.g. for Ubuntu 22.04 runners the list can be found [here](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md)). ROOT is not pre-installed, so we will have to add a step to install it ourselves. After reading the [ROOT documentation](https://root.cern/install/#run-in-a-docker-container), we find that a convenient way to run it on various systems is using something called a Docker container.
+
+There are several tools that are used for containerization, like Docker, Podman, and Apptainer (formerly Singularity). For this tutorial you don't need to know anything about containerization. You can just think of this as the base software set that comes pre-installed on the system that runs your code.
+
+We will be using the Docker images hosted at the [`rootproject/root` Docker Hub](https://hub.docker.com/r/rootproject/root). Let's start by using the image with tag `6.26.10-conda`.
 
 ```yaml
 build_skim:
   runs-on: ubuntu-latest
-  defaults:
-    run:
-      shell: bash -el {0}
+  container: rootproject/root:6.26.10-conda
   steps:
-    - name: Install ROOT
-      uses: mamba-org/setup-micromamba@v1
-      with:
-        environment-name: env
-        create-args: root
+    - name: checkout repository
+      uses: actions/checkout@v4
     - name: build
       run: |
         COMPILER=$(root-config --cxx)
         $COMPILER -g -O3 -Wall -Wextra -Wpedantic -o skim skim.cxx
 ```
 
+Note the extra line `container: rootproject/root:6.26.10-conda` that specifies the container image that we want to use. Since it comes pre-packages with ROOT we no longer need to have a step to install it. This image also contains other tools that we will need for the rest of the tutorial.
```

Comment:
```suggestion
Note the extra line `container: rootproject/root:6.26.10-conda` that specifies the container image that we want to use. Since it comes pre-packaged with ROOT, we do not need to have a step to install it. This image also contains other tools that we will need for the rest of the tutorial.
```